.pre-commit-config.yaml

```diff
@@ -1,8 +1,33 @@
 exclude: '^$'
 fail_fast: false
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v2.2.3
     hooks:
-    - id: flake8
-      additional_dependencies: [flake8-docstrings, flake8-debugger, flake8-bugbear]
+      - id: flake8
+        additional_dependencies: [flake8-docstrings, flake8-debugger, flake8-bugbear]
+        # TODO: Add mccabe to the above
+        args: [--max-complexity, "10"]
+  - repo: https://github.com/PyCQA/bandit
+    rev: '1.7.0' # Update me!
+    hooks:
+      - id: bandit
+        args: [-x, satpy/tests]
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.910-1'  # Use the sha / tag you want to point at
+    hooks:
+      - id: mypy
+        additional_dependencies:
+          - types-docutils
+          - types-pkg-resources
+          - types-PyYAML
+          - types-requests
+  - repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+        language_version: python3
```

Comment:
What do those commit hooks do exactly?  I suppose it's two more development dependencies.  How do they complement or replace tests currently run in CI?  Do they edit files?  Rewriting commits or adding new commits?  Do they enforce any rules that aren't written either in PEP8 or in Satpy developer guides?

Reply:
Bandit, mypy, and flake8 will only warn about issues. isort is the only one that will rewrite the file. If any of these fail the commit is not done and you have to reattempt the commit. If isort is the only one that failed (meaning it reordered the imports) then in most cases you can just reattempt the commit right away (assuming import reorder doesn't break things).